core/src/main/scala/kafka/zk/ZkData.scala

```diff
@@ -293,14 +297,17 @@ object TopicZNode {
         removingReplicasAssignmentJson += (partition.partition.toString -> replicaAssignment.removingReplicas.asJava)
     }
 
-    Json.encodeAsBytes(Map(
+    val topicAssignment = mutable.Map(
       "version" -> 2,
```

Comment:
Why not bump the version? the version is bumped to 3 according to the KIP

Reply:
Good catch. I'll fix that.